src/hd/hd.c

```diff
@@ -3150,42 +3150,25 @@ int hd_is_sgi_altix(hd_data_t *hd_data)
 }
 
 
-
-
 /*
  * check for xen hypervisor
  */
 int hd_is_xen(hd_data_t *hd_data)
 {
 #if defined(__i386__) || defined(__x86_64__)
 
-  char signature[13];
-  unsigned u, foo;
+  unsigned eax, ebx, ecx, edx;
 
   __asm__(
-#ifdef __i386__
-    "push %%ebx\n\t"
-    "cpuid\n\t"
-    "mov %%ebx,(%%esi)\n\t"
-    "mov %%ecx,4(%%esi)\n\t"
-    "mov %%edx,8(%%esi)\n\t"
-    "pop %%ebx"
-#else
-    "push %%rbx\n\t"
-    "cpuid\n\t"
-    "mov %%ebx,(%%rsi)\n\t"
-    "mov %%ecx,4(%%rsi)\n\t"
-    "mov %%edx,8(%%rsi)\n\t"
-    "pop %%rbx"
-#endif
-    : "=a" (u), "=c" (foo)
-    : "a" (0x40000000), "c" (0), "S" (signature)
-    : "%edx"
+    "cpuid"
```

Comment:
I would like to see link to that instruction explanation here beside code. So it will be easier to find then in pull request description.